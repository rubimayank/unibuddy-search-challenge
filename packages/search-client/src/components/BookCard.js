import React from 'react';
import styled from 'styled-components';
import {
  MdClose
} from 'react-icons/md';

import Card from './Card';
import Button from './BaseButton';

const Title = styled.h3`
  font-size: 18px;
  padding-right: 40px;
`;

const Author = styled.span`
  font-size: 14px;
  color: #888;
`;

const Summary = styled.div`
  font-size: 14px;
  line-height: 20px;
`;

const Remove = styled(Button)`
  position: absolute;
  right: 0;
  top: 0;
  padding: 8px;
`;

function BookCard({ book, onRemove }) {
  const { author, title, summary } = book;
  const shortSummary = summary.substring(0, 160);
  return (
    <Card>
      <div>
        <Title>{title}</Title>
        <Author>{author}</Author>
      </div>
      <Summary>
        {shortSummary}
        {summary.length > 200 && '...'}
      </Summary>
      {onRemove && (
        <Remove onClick={onRemove} title='Remove'>
          <MdClose />
        </Remove>
      )}
    </Card>
  );
}

export default BookCard;
