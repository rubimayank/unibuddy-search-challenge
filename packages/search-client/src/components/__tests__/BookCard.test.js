import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import BookCard from '../BookCard';

const book = {
    title: 'title',
    author: 'author',
    summary: 'summary',
};

const lorem = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas pulvinar quam id augue porta vestibulum in vel justo. Cras maximus volutpat magna et sagittis. Sed lacinia, tortor vel maximus ullamcorper, leo felis posuere nisi, sed viverra sem erat non orci. Etiam semper dui ac venenatis malesuada. Donec vitae turpis rhoncus, egestas nisi id, porta quam.';

describe('BookCard', () => {
  it('it should render', () => {
    const snap = renderer.create(<BookCard book={book} />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('it should render long summary with elipses', () => {
    const longSummaryBook = {
      ...book,
      summary: lorem,
    };
    const snap = renderer.create(<BookCard book={book} />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('it should render with remove button', () => {
    const snap = renderer
      .create(<BookCard book={book} onRemove={() => {}} />)
      .toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('should call onRemove', () => {
    let removed = false;
    const wrapper = renderer
      .create(<BookCard book={book} onRemove={()=>{ removed = true; }} />)
      .root;
    const removeBtn = wrapper.findByType('button');
    removeBtn.props.onClick();
    expect(removed).toEqual(true);
  });
});
