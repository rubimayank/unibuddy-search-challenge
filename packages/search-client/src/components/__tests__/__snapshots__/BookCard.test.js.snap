// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard it should render 1`] = `
.c0 {
  position: relative;
  padding: 12px;
  box-shadow: rgba(0,0,0,0.12) 1px 1px 1px 2px;
  border-radius: 6px;
  margin: 8px;
}

.c0 > div {
  padding-bottom: 12px;
}

.c0 > div:last-child {
  padding-bottom: 0;
}

.c1 {
  font-size: 18px;
  padding-right: 40px;
}

.c2 {
  font-size: 14px;
  color: #888;
}

.c3 {
  font-size: 14px;
  line-height: 20px;
}

<div
  className="c0"
>
  <div>
    <h3
      className="c1"
    >
      title
    </h3>
    <span
      className="c2"
    >
      author
    </span>
  </div>
  <div
    className="c3"
  >
    summary
  </div>
</div>
`;

exports[`BookCard it should render long summary with elipses 1`] = `
.c0 {
  position: relative;
  padding: 12px;
  box-shadow: rgba(0,0,0,0.12) 1px 1px 1px 2px;
  border-radius: 6px;
  margin: 8px;
}

.c0 > div {
  padding-bottom: 12px;
}

.c0 > div:last-child {
  padding-bottom: 0;
}

.c1 {
  font-size: 18px;
  padding-right: 40px;
}

.c2 {
  font-size: 14px;
  color: #888;
}

.c3 {
  font-size: 14px;
  line-height: 20px;
}

<div
  className="c0"
>
  <div>
    <h3
      className="c1"
    >
      title
    </h3>
    <span
      className="c2"
    >
      author
    </span>
  </div>
  <div
    className="c3"
  >
    summary
  </div>
</div>
`;

exports[`BookCard it should render with remove button 1`] = `
.c0 {
  position: relative;
  padding: 12px;
  box-shadow: rgba(0,0,0,0.12) 1px 1px 1px 2px;
  border-radius: 6px;
  margin: 8px;
}

.c0 > div {
  padding-bottom: 12px;
}

.c0 > div:last-child {
  padding-bottom: 0;
}

.c1 {
  font-size: 18px;
  padding-right: 40px;
}

.c2 {
  font-size: 14px;
  color: #888;
}

.c3 {
  font-size: 14px;
  line-height: 20px;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  overflow: hidden;
  padding: 8px 16px;
  margin: 2px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  text-align: center;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  background: none;
  border: 0 none;
  border-radius: 2px;
  font-size: 14px;
  line-height: 1.3;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  right: 0;
  top: 0;
  padding: 8px;
}

.c4:hover {
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  opacity: .85;
}

.c4:active {
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  opacity: .75;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
  opacity: .50;
}

.c4:disabled:hover {
  opacity: .50;
}

<div
  className="c0"
>
  <div>
    <h3
      className="c1"
    >
      title
    </h3>
    <span
      className="c2"
    >
      author
    </span>
  </div>
  <div
    className="c3"
  >
    summary
  </div>
  <button
    className="sc-AxirZ c4"
    onClick={[Function]}
    title="Remove"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
</div>
`;
